api/data.go

```diff
@@ -180,3 +200,12 @@ func isHexadecimal(s string) bool {
 func isLower(s string) bool {
 	return strings.ToLower(s) == s
 }
+
+func sendJSONError(ctx context.Context, w http.ResponseWriter, code int, message string) {
+	w.WriteHeader(code)
+	if err := json.NewEncoder(w).Encode(map[string]string{"error": message}); err != nil {
+		log.Error(ctx, "error encoding results to JSON", err)
```

Comment:
What about "error encoding error message to JSON"?

Reply:
done it now